app/controllers/openid_application_controller.rb

```diff
@@ -137,15 +137,22 @@ def establish_session(profile)
     @session = build_session(ttl,
                              Okta::UserProfile.new({ 'last_login_type' => profile['last_login_type'],
                                                      'SecID' => profile['SecID'], 'VistaId' => profile['VistaId'],
-                                                     'npi' => profile['npi'], 'icn' => profile['icn'],
-                                                     'uuid' => uuid(profile) }))
+                                                     'npi' => profile['npi'], 'icn' => profile['icn'] }))
     @session.save && user_identity.save && @current_user.save
   end
 
-  # Helper method that uses the profile uuid set by SSOe since the sub == ICN in that scenario
-  # but falls back to the token.identifiers.uuid
-  def uuid(profile)
-    profile['uuid'] || token.identifiers.uuid
+  # Ensure the Okta profile ICN continues to match the MPI ICN
+  # If mismatched, revoke in Okta, set @session to nil, and return false
+  # POA support (profile['icn'].nil?)
+  def confirm_icn_match(profile)
+    if profile['icn'].nil? || @current_user.icn == profile['icn']
+      true
+    else
+      Okta::Service.new.clear_user_session(token.identifiers.okta_uid)
+      @session = nil
+      log_message_to_sentry('Profile ICN mismatch detected. Clearing user session.', :error)
```

Comment:
What is `:error` here?

Reply:
Log level.  I went back and forth between error and warn since it is a scenario we are designing to handle vs an exception.  But the ICN mismatch will return a 401, so thought that more aligned with an error.

Reply:
Switching to log level `:warn` now that this will detect and log to sentry only, not actually clear the Okta user session.